packages/edit-navigation/src/components/layout/use-menus.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+
+export default function useMenus() {
+	const wpMenus = useSelect( ( select ) => select( 'core' ).getMenus() );
+
+	const [ currentMenu, setCurrentMenu ] = useState( 0 );
+	const [ menus, setMenus ] = useState( null );
+
+	useEffect( () => {
+		if ( wpMenus?.length ) {
+			setMenus( wpMenus );
+			setCurrentMenu( wpMenus[ 0 ].id );
+		}
+	}, [ wpMenus ] );
```

Comment:
@draganescu As mentioned in https://github.com/WordPress/gutenberg/issues/22340#issuecomment-628531823, I don't think this code will work, as the deleted menus get restored whenever `wpMenus` changes.

This effect is listening for changes to `wpMenus`, and whenver that happens `setMenus( wpMenus )` on line 15 effectively undoes any of the changes to `menus` performed by the `onDeleteMenu` callback.

Reply:
It does currently work and it fixes the problem in #22340, but I agree that keeping a list of `deletedItems` is better on the long run.

Reply:
Yeah, after further debugging I noticed it does usually work, I think it relies on the fact that typically when `wpMenus` changes, the entity cache has been cleared and so deleted items are no longer present in `wpMenus`.

Reply:
+1, this also affects the menu items themselves - we need one solution to rule them all :-)